connectors/tests/test_sync.py

```diff
@@ -658,3 +669,16 @@ async def _sync(*args):
         Connector.sync = old_sync
 
     patch_logger.assert_instance(Exception)
+
+
+@pytest.mark.asyncio
+async def test_concurrent_syncs(mock_responses, patch_logger, set_env):
+    await set_server_responses(mock_responses, [FAKE_CONFIG, FAKE_CONFIG, FAKE_CONFIG])
+    service = create_service(CONFIG_FILE)
+    asyncio.get_event_loop().call_soon(service.stop)
+    await service.run()
+
+    # make sure we synced the three connectors
+    patch_logger.assert_present("[1] Sync done: 1 indexed, 0  deleted. (0 seconds)")
+    patch_logger.assert_present("[2] Sync done: 1 indexed, 0  deleted. (0 seconds)")
+    patch_logger.assert_present("[3] Sync done: 1 indexed, 0  deleted. (0 seconds)")
```

Comment:
Is it possible to do assertion on mocks/storage instead of logger?

Reply:
Sure why not

This assertion makes sure that we reached that line and did index a document https://github.com/elastic/connectors-python/blob/main/connectors/byoc.py#L553

All calls to Elasticsearch are mocked / what do you have in mind to reach the same level of coverage+assertion?

Reply:
For me good rule of thumb for testing is: if you delete some lines from the project and breaking the logic but the test will still pass, then test should be improved.

In essence in this case it might mean checking elasticsearch mocked requests that indicate the sync was finished, for example. Or mocking more classes and checking they were called.

Reply:
Something that we'll need to follow-up upon structurally, let's do it later, I'll prepare PRs/slides to discuss